clients/adapter-kafka/src/test/java/org/eclipse/hono/adapter/client/telemetry/kafka/AbstractKafkaBasedDownstreamSenderTest.java

```diff
@@ -335,20 +335,52 @@ public void testThatCreationTimeIsAddedWhenNotPresentAndTtdIsSet(final VertxTest
 
     }
 
+    /**
+     * Verifies that if the properties contain a <em>ttl</em> property but no <em>creation-time</em> then the later is
+     * added.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testThatCreationTimeIsAddedWhenNotPresentAndTtlIsSet(final VertxTestContext ctx) {
+        final MockProducer<String, Buffer> mockProducer = TestHelper.newMockProducer(true);
+        final AbstractKafkaBasedDownstreamSender sender = newSender(TestHelper.newProducerFactory(mockProducer));
+
+        // GIVEN properties that contain a TTL
+        final long ttl = 99L;
+        final Map<String, Object> properties = new HashMap<>();
+        properties.put("ttl", ttl);
+
+        // WHEN sending the message
+        sender.send(topic, tenant, device, qos, CONTENT_TYPE, null, properties, null)
+                .onComplete(ctx.succeeding(t -> {
+                    ctx.verify(() -> {
+                        // THEN the producer record contains a creation time
+                        final ProducerRecord<String, Buffer> record = mockProducer.history().get(0);
+                        assertThat(record.headers())
+                                .containsOnlyOnce(new RecordHeader("ttl", Json.encode(ttl).getBytes()));
+                        assertThat(record.headers().headers("creation-time")).isNotNull();
```

Comment:
is this related to the ttl property?

Reply:
Yes. Are you asking about the relevance of the `creation-time`? If a `ttl` header is present, we also need the `creation-time` header. `AbstractKafkaBasedDownstreamSender` adds the latter if it's missing. This is what the test verifies.

Reply:
Oh yes, I saw that. There also was a comment whether the standard Kafka creation time property should be used (instead of the _creation-time_ property?).

Reply:
Oh, yes the TODO :) From what I know, there is quite a bit of complexity in Kafka's handling of message timestamps (behavior can be configured at the topic, timestamps might be changed by the broker in some cases, enabling/disabling of compression affects how timestamps are treated [and so on](https://cwiki.apache.org/confluence/display/KAFKA/KIP-32+-+Add+timestamps+to+Kafka+message). So I tend to use our own header for Hono's messaging and not rely on the message timestamp.